include/pgduckdb/duckdb_vendor/enum_type_info_templated.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "duckdb/common/extra_type_info.hpp"
+#include "duckdb/common/vector.hpp"
+#include "duckdb/common/types/vector.hpp"
+#include "duckdb/common/string_map_set.hpp"
+#include "duckdb/common/serializer/deserializer.hpp"
+
+// This is copied directly without changes from 'duckdb/src/common/extra_type_info.cpp'
+// The reason this is copied is to be able to inherit from it.
```

Comment:
```suggestion
// The reason this is copied is to be able to inherit from it.
// This can be removed once DuckDB 1.2.0 is released, which contains this PR: 
// https://github.com/duckdb/duckdb/pull/14038
```